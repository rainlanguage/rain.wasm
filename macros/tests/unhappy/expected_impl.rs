#[macro_use]
extern crate wasm_bindgen_utils_macros;

#[wasm_export]
struct TestStruct;

#[wasm_export]
enum TestEnum {}

#[wasm_export]
fn test() {}

#[wasm_export]
type SomeType = u8;

#[wasm_export]
mod SomeMod {}

#[wasm_export]
const SOME_COST: u8 = 1;

#[wasm_export]
static SOME_STATIC: u8 = 1;

fn main() {}
