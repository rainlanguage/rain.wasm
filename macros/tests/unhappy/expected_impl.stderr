error: expected `impl`, wasm_export macro is only applicable to impl blocks
 --> tests/unhappy/expected_impl.rs:5:1
  |
5 | struct TestStruct;
  | ^^^^^^

error: expected `impl`, wasm_export macro is only applicable to impl blocks
 --> tests/unhappy/expected_impl.rs:8:1
  |
8 | enum TestEnum {}
  | ^^^^

error: expected `impl`, wasm_export macro is only applicable to impl blocks
  --> tests/unhappy/expected_impl.rs:11:1
   |
11 | fn test_fn() {}
   | ^^

error: expected `impl`, wasm_export macro is only applicable to impl blocks
  --> tests/unhappy/expected_impl.rs:14:1
   |
14 | type TestType = u8;
   | ^^^^

error: expected `impl`, wasm_export macro is only applicable to impl blocks
  --> tests/unhappy/expected_impl.rs:17:1
   |
17 | mod test_mod {}
   | ^^^

error: expected `impl`, wasm_export macro is only applicable to impl blocks
  --> tests/unhappy/expected_impl.rs:20:1
   |
20 | const TEST_COST: u8 = 1;
   | ^^^^^

error: expected `impl`, wasm_export macro is only applicable to impl blocks
  --> tests/unhappy/expected_impl.rs:23:1
   |
23 | static TEST_STATIC: u8 = 1;
   | ^^^^^^
